go/engine/login_provision.go

```diff
@@ -748,29 +749,50 @@ func (e *loginProvision) chooseDevice(m libkb.MetaContext, pgp bool) (err error)
 	if len(id) == 0 {
 		// they chose not to use a device
 		m.Debug("user has devices, but chose not to use any of them")
-		if pgp {
-			if hasPUK {
-				return libkb.ProvisionViaDeviceRequiredError{}
-			}
 
+		if pgp && !hasPUK {
 			// they have pgp keys, so try that:
-			if err := e.tryPGP(m); err != nil {
-				// TODO CORE-10630: consider this flow with autoreset
-				return err
+			err = e.tryPGP(m)
+			if err == nil {
+				// Provisioning succeeded
+				return nil
 			}
 
-			return nil
+			// If autoreset is enabled, an error here should passthrough into
+			// autoreset.
+			if autoresetEnabled {
+				m.Warning("Unable to log in with a PGP signature: %s", err.Error())
+			} else {
+				return err
+			}
 		}
 
-		if !pipelineEnabled {
+		// Error differs depending on the input.
+		if !autoresetEnabled {
+			if pgp && hasPUK {
+				return libkb.ProvisionViaDeviceRequiredError{}
+			}
 			return libkb.ProvisionUnavailableError{}
 		}
 
-		// TODO CORE-10631 make this a UI
-		m.G().Log.Info(`
-The only way to provision this device is with access to one of your existing
-devices. You can try again later, or if you have lost access to all your
-existing devices you can reset your account and start fresh:`)
+		// Prompt the user whether they'd like to enter the reset flow.
+		// We will ask them for a password in AccountReset.
+		enterReset, err := m.UIs().LoginUI.PromptResetAccount(m.Ctx(), keybase1.PromptResetAccountArg{
+			Text: "The only way to provision this device is with access to one of your existing " +
+				"devices. You can try again later, or if you have lost access to all your " +
+				"existing devices you can reset your account and start fresh. Would you " +
+				"like to request a reset of your account?",
+		})
+		if err != nil {
+			return err
+		}
+
+		if !enterReset {
+			// Although we didn't enter the pipeline, we don't want the flow to try logging in,
+			// so we mark it as if we have just joined the pipeline.
+			e.resetPending = true
```

Comment:
can you make a second state variable and then expose a method on the engine which the callers use to determine if there is a pending reset or if the reset was skipped? there's not need to overload this one variable, I think it will be confusing in the long run.

Reply:
I've renamed resetPending and created methods exporting it on engines that get queried on it. wdyt now?